source/connect.txt

```diff
@@ -46,39 +46,96 @@ the relevant values for your MongoDB deployment.
 
 .. include:: /includes/usage-examples/sample-app-intro.rst
 
-.. literalinclude:: /includes/usage-examples/connect-sample-app.py
-   :language: python
-   :copyable: true
-   :linenos:
-   :emphasize-lines: 4-6
+Select the :guilabel:`Synchronous` or :guilabel:`Asynchronous` tab to see the corresponding
+code:
+
+.. tabs::
+
+   .. tab:: Synchronous
+      :tabid: sync
+
+      .. literalinclude:: /includes/usage-examples/connect-sample-app.py
+         :language: python
+         :copyable: true
+         :linenos:
+         :emphasize-lines: 4-6
+
+   .. tab:: Asynchronous
+      :tabid: async
+
+      .. literalinclude:: /includes/usage-examples/connect-sample-app-async.py
+         :language: python
+         :copyable: true
+         :linenos:
+         :emphasize-lines: 6-8
 
 Connection
 ----------
 
 Local Deployment
 ~~~~~~~~~~~~~~~~
 
-.. code-block:: python
+.. tabs::
```

Comment:
S: include introductions about what's happening in the code examples (applies to all)
https://www.mongodb.com/docs/languages/kotlin/kotlin-sync-driver/current/connect/
has example text